.github/workflows/rust.yml

```diff
@@ -0,0 +1,22 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build
+      run: cd bestool && cargo build --verbose
```

Comment:
Only thing I'd add is that you could use `actions-rs/cargo`, but IIRC last time I checked, they don't support subdirectories. One way around this is using a fork of that action, or sticking with this approach. But you could use `working-directory` as a `with:` key, so you don't have to cd.